k8s/kubevirt/virtualmachine.go

```diff
@@ -250,3 +275,35 @@ func (c *Client) GetVMConfigMaps(vm *kubevirtv1.VirtualMachine) []string {
 	}
 	return configMaps
 }
+
+// IsVMAgentConnected returns true if Running VM has qemu-guest-agent running in the guest OS
+func (c *Client) IsVMAgentConnected(vm *kubevirtv1.VirtualMachine) bool {
+
+	if c.IsVirtualMachineRunning(vm) {
+		for _, cond := range vm.Status.Conditions {
+			if cond.Type == "AgentConnected" {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// GetVMFreezeRule returns freeze Rule action for given VM
+func (c *Client) GetVMFreezeRule(vm *kubevirtv1.VirtualMachine) string {
+
+	return fmt.Sprintf(VMFreezeCmd, vm.GetName(), vm.GetNamespace())
+}
+
+// GetVMUnFreezeRule returns unfreeze Rule action for given VM
+func (c *Client) GetVMUnFreezeRule(vm *kubevirtv1.VirtualMachine) string {
+
+	return fmt.Sprintf(VMUnFreezeCmd, vm.GetName(), vm.GetNamespace())
+}
+
+// GetVMPodLabel return podSelector label for the given VM
+func (c *Client) GetVMPodLabel(vm *kubevirtv1.VirtualMachine) map[string]string {
+	return map[string]string{
+		VMPodSelectorLabel: vm.GetName(),
+	}
+}
```

Comment:
Do we need these functions in sched-ops? I was thinking these are not using the receiver type `(c *Client)` and also these can be just a helper function in Torpedo itself right?
Shouldn't we be writing only those function here which use the kubevirt client?

Reply:
For phase 2, px-backup will consume this.
For regression cases with phase1 automation and for namespace backup scenario testing even for phase 2 test automation and integration test automation torpedo and px-backup will use it
Since this is tightly coupled with VM and signature should be same when called by any of them. Its better we use common function. Hence sched-ops seem to be the right place holder even though we are not making client call.

Reply:
For phase 2, px-backup will consume this.
For regression cases with phase1 automation and for namespace backup scenario testing even for phase 2 test automation and integration test automation torpedo and px-backup will use it
Since this is tightly coupled with VM and signature should be same when called by any of them. Its better we use common function. Hence sched-ops seem to be the right place holder even though we are not making client call.

Reply:
My view on this is for the following calls should they be in sched-ops or Torpedo? As these don't really need a client hanlder? We can discuss on this and see whats the best place to add this

GetVMFreezeRule()
 GetVMPodLabel()
 GetVMUnFreezeRule()

Following can be in sched-ops
IsVMAgentConnected()